### PR TITLE
Ownership `... on Team` GraphQL resolvers

### PR DESCRIPTION
**Pull request by cbart**
Part of #47980

Introduce team resolution in graphql.

## Test plan

Add a component test resolving a team - goes down to fake database.

Also, it actually works:
https://www.loom.com/share/a3457e575b0041bdb6d304ae7227db81

### CODE REVIEW

**Review by leonore — APPROVED** [review]

File: cmd/frontend/graphqlbackend/teams_test.go [hunk 1]
```diff
@@ -1386,11 +1386,6 @@ func TestMembersSearch(t *testing.T) {
 			UserID: userID,
 		})
 	}
-	idOfMissingUser := -7
-	fs.AddTeamMember(&types.TeamMember{
-		TeamID: team.ID,
-		UserID: int32(idOfMissingUser),
-	})
```
Comment (cbart) [inline cmd/frontend/graphqlbackend/teams_test.go:1393]: The test here needed to change due to implementing proper user/teams not-found semantics (returning a not found err as opposed to nil, nil).

This case is actually inconsistent data, not sure why I put it here.

File: enterprise/cmd/frontend/internal/own/resolvers/resolvers_test.go [hunk 2]
```diff
@@ -60,6 +63,26 @@ func (s fakeOwnService) ResolveOwnersWithType(_ context.Context, owners []*codeo
...
+
+func (g fakeGitserver) ReadFile(_ context.Context, _ authz.SubRepoPermissionChecker, repoName api.RepoName, commitID api.CommitID, file string) ([]byte, error) {
+	if g.files == nil {
+		return nil, os.ErrNotExist
+	}
+	content, ok := g.files[repoPath{Repo: repoName, CommitID: commitID, Path: file}]
+	if !ok {
+		return nil, os.ErrNotExist
+	}
+	return []byte(content), nil
```
Comment (cbart) [inline enterprise/cmd/frontend/internal/own/resolvers/resolvers_test.go:85]: I'm thinking, we could use a shared fake git server implementation.

Comment (eseliger) [reply enterprise/cmd/frontend/internal/own/resolvers/resolvers_test.go:85]: there's `gitserver.NewMockClient()` I think. Could that help?

**Review by eseliger — APPROVED** [review]

File: enterprise/cmd/frontend/internal/own/resolvers/resolvers_test.go (hunk 2 above)
Comment (cbart) [reply enterprise/cmd/frontend/internal/own/resolvers/resolvers_test.go:87]: I was thinking more of having this map implementation, which seems very convenient.

**Pull request merged**